### PR TITLE
Migrate tests to jest

### PR DESCRIPTION
**Pull request by PRTTMPRPHT**
Since #183 has been open for quite a while, I thought I'd do the migration from mocha, sinon and chai to jest. :)

Changes made:

- Migrate all tests to work with jest
- Clean up test dependencies
- Added coverage reporting
- Merge previously browser-based tests with the jsdom-based tests into one test run

Caveats:

Since jest doesn't really support browser-based testing, a mock plugin for canvases has been included. Since that is not a real canvas implementation, values provided by measureText will be off by a bit in the tests. This shouldn't really be an issue, but has to be kept in mind when writing these kinds of tests. I tried to preserve the original test logic in the concerned tests.

### CODE REVIEW

**Comment by ekh64** [comment]
This is INCREDIBLE. I'll be sure to review this sometime this week. Thank you so much for your hard work

**Comment by PRTTMPRPHT** [comment]
No problem. :) Lemme know should you need any changes. Have a great week!

**Comment by ekh64** [comment]
@PRTTMPRPHT Just a quick update, looks like our primary maintainer is on vacation this week. I can certainly review, but I'd prefer to wait for him to return before merging.

**Comment by PRTTMPRPHT** [comment]
@ekh64 Sure, fine with me!

File: tests/spec/XAxisLabels.spec.js
Comment (ekh64) [inline tests/spec/XAxisLabels.spec.js:110]: Thanks for the explanation!

**Review by ekh64 — APPROVED** [review]
This looks great! I'll make sure @iezer has a look when he's back. Hopefully we can get this merged next week 🎉

File: tests/browser/index.js
Comment (iezer) [inline tests/browser/index.js:12]: I'm a fan of having some browser-based tests for these reasons. It's easy to debug in the Chrome debugger. Also since this is a dataviz library that interacts with canvas a lot it's better not to have used the mocked and inaccurate version of measureText. What do you think about keeping the browser based tests using mocha for now and reducing this PR to only migrate the JSDOM tests to jest for now. We can revisit the browser based tests later on. @PRTTMPRPHT and @ekh64

File: tests/spec/Bar.spec.js
Comment (iezer) [inline tests/spec/Bar.spec.js:64]: 👍 It's a big improvement using jest instead of mocha in these jsdom tests. Thanks!

**Review by iezer — CHANGES_REQUESTED** [review]
Thanks for taking this on @PRTTMPRPHT ! We really appreciate it. How did you find out about this library? Have you used it before?

File: tests/browser/index.js
Comment (PRTTMPRPHT) [reply tests/browser/index.js:12]: Yeah, I can understand that! Then I'll revert the changes for the concerned tests and add back the dependencies. I'm currently on vacation, so I will do that once I'm back on July 1st.

**Comment by PRTTMPRPHT** [comment]
@iezer You're welcome! I've recently been researching projects to contribute to, as I want to get more into open-source work. I've neglected this in my life as a developer for far too long :) Since I've done a lot with dataviz in the past this one was a natural pick. Haven't yet used it in a project but it's always good to have some alternatives to the usual bunch available. So if you need a helping hand for something else, please be sure to let me know!

**Comment by PRTTMPRPHT** [comment]
@iezer There you go! I restored the browser-based tests to their original state.

**Comment by iezer** [comment]
@PRTTMPRPHT Thanks for contributing. We really appreciate it.

**Comment by dandelany** [comment]
Y'all are awesome, thanks for carrying the Reactochart torch @iezer @ekh64 @PRTTMPRPHT  :) ❤️

**Pull request merged**
